nixos/modules/services/misc/mollysocket.nix

```diff
@@ -0,0 +1,105 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mollysocket;
+  configuration = format.generate "mollysocket.conf" cfg.settings;
+  format = pkgs.formats.toml { };
+in {
+  options.services.mollysocket = {
+    enable = mkEnableOption (mdDoc ''
+      [MollySocket](https://github.com/mollyim/mollysocket) for getting Signal
+      notifications via UnifiedPush
+    '');
+
+    settings = mkOption {
+      default = { };
+      description = mdDoc ''
+        Configuration for MollySocket. Available options are listed
+        [here](https://github.com/mollyim/mollysocket#configuration).
+      '';
+      type = types.submodule {
+        freeformType = format.type;
+        options = {
```

Comment:
We should probably also expose an option for `RUST_LOG`, but thats just a minor nit pick

Reply:
If you want to set `RUST_LOG` use `systemd.services.mollysocket.serviceConfig.Environment`.

Reply:
I'd consider that a bad user experience

Reply:
But setting `RUST_LOG` is not necessary for normal operation.

Reply:
Should we make `RUST_LOG=info` the default?

Reply:
> But setting RUST_LOG is not necessary for normal operation.

Right, I think debugging the service is something we should support without messing around with the service config directly however.

> Should we make RUST_LOG=info the default?

I'd also be in favor of that, its the default in the upstream service definition.